app/controllers/user.js

```diff
@@ -189,22 +203,38 @@ module.exports = {
 
   // GET INVITE LINK
   getInviteLink: async (req, res, next) => {
-    const token = jwt.sign({ _id: req.user._id, expiry: Date.now() + 24 * 3600 * 1000 }, process.env.JWT_SECRET)
-    const inviteLink = `${req.protocol}://${req.get('host')}/user/invite/${token}`
-    return res.status(HttpStatus.OK).json({ inviteLink: inviteLink })
+    try {
+      const { role } = req.query
+      const token = jwt.sign({ _id: req.user._id, role: role, expiry: Date.now() + 24 * 3600 * 1000 }, process.env.JWT_SECRET)
+      const inviteLink = `${req.protocol}://${req.get('host')}/user/invite/${token}`
+      return res.status(HttpStatus.OK).json({ inviteLink: inviteLink })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
   },
 
   // PROCESS THE INVITE LINK
   processInvite: async (req, res, next) => {
-    const { token } = req.params
-    const decodedToken = jwt.verify(token, process.env.JWT_SECRET)
-    // check if token not expired and sender exist in db then valid request
-    const user = await User.findById(decodedToken._id)
-    if (user && Date.now() <= decodedToken.expiry) {
-      console.log('Valid invite!')
-      return res.status(HttpStatus.OK).json({ success: true, msg: 'Redirect user to register in client side!' })
+    try {
+      const { token } = req.params
+      const decodedToken = jwt.verify(token, process.env.JWT_SECRET)
+      // check if token not expired and sender exist in db then valid request
+      const user = await User.findById(decodedToken._id)
+      if (user && Date.now() <= decodedToken.expiry) {
+        console.log('Valid invite!')
+        if (decodedToken.role === 'user') {
+          // TODO: CHANGE THE URL IN PRODUCTION
+          return res.redirect('http://localhost:3000/')
```

Comment:
please dont hardcode the URL. We can have a variable in env `apibaseurl` and then consume in place of the hardcoded url.